dbt-snowflake/src/dbt/adapters/snowflake/impl.py

```diff
@@ -432,15 +455,47 @@ def debug_query(self):
 
     @classmethod
     def _get_adapter_specific_run_info(cls, config: RelationConfig) -> Dict[str, Any]:
-        table_format: Optional[str] = None
-        if (
-            config
-            and hasattr(config, "_extra")
-            and (relation_format := config._extra.get("table_format"))
-        ):
-            table_format = relation_format
-
-        return {
-            "adapter_type": "snowflake",
-            "table_format": table_format,
+        # `config` is not a RelationConfig!
+        run_info = {
+            "adapter_type": constants.ADAPTER_TYPE,
+            "table_format": None,
         }
+
+        if config and hasattr(config, "_extra"):
+
+            catalog = config._extra.get("catalog")
+
+            if _table_format := config._extra.get("table_format"):  # type:ignore
+                run_info["table_format"] = _table_format
+            elif not catalog:
+                # no table_format and no catalog definitely means native table
+                run_info["table_format"] = constants.NATIVE_TABLE_FORMAT
+            elif catalog == constants.DEFAULT_NATIVE_CATALOG.name:  # type:ignore
+                # if the user happens to set the catalog to the native, catch that
+                run_info["table_format"] = constants.NATIVE_TABLE_FORMAT
+            else:  # catalog is set, and it's not the native catalog
+                # it's unlikely that users will set a catalog that's not Iceberg
+                run_info["table_format"] = constants.ICEBERG_TABLE_FORMAT
+
+        return run_info
+
+    @available
+    def build_catalog_relation(self, model: RelationConfig) -> Optional[CatalogRelation]:
+        """
+        Builds a relation for a given configuration.
+
+        This method uses the provided configuration to determine the appropriate catalog
+        integration and config parser for building the relation. It defaults to the managed Iceberg
+        catalog if none is provided in the configuration for backward compatibility.
+
+        Args:
+            model (RelationConfig): `config.model` (not `model`) from the jinja context
+
+        Returns:
+            Any: The constructed relation object generated through the catalog integration and parser
+        """
+        if catalog := parse_model.catalog_name(model):
+            catalog_integration = self.get_catalog_integration(catalog)
+            relation = catalog_integration.build_relation(model)
```

Comment:
could just return this

Reply:
Agreed. I think this was left from debugging where I wanted to look at the relation before returning it.